compaction.go

```diff
@@ -267,11 +303,20 @@ func (c *compaction) newInputIter(
 				rangeDelIter = nil
 			}
 		}
+		if rangeDelIter != nil {
+			// Truncate the range tombstones returned by the iterator to the upper
+			// bound of the atomic compaction unit.
+			if upperBound := c.atomicCompactionUnitUpperBound(f); upperBound != nil {
```

Comment:
Cool, I didn't think of computing this as the compaction is ongoing.

Reply:
Yeah. We could precompute this in `newInputIter` and store the results in a map keyed by `*fileMetadata`, but that seems like overkill when compactions usually involve only a few inputs.